explorer/components/Era5WrfXrayWrapper.vue

```diff
@@ -0,0 +1,195 @@
+      if (element && element.hasChildNodes()) {
+        $Plotly.purge(chartId)
+      }
+    } catch (error) {
+      // Ignore purge errors - chart may already be cleaned up
+      console.debug(`Chart purge skipped for ${chartId}:`, error)
+    }
+  })
+}
+
+const fetchData = () => {
+  if (!latLng.value) return
+  dataStore.fetchData(endpoint)
+}
+
+onMounted(() => {
+  fetchData()
+})
+
+watch(latLng, async () => {
+  // Proactively purge all charts before data changes (following WetDaysPerYear pattern)
+  purgeAllCharts()
+  dataStore.apiData[endpoint] = null
+  dataStore.dataErrors[endpoint] = false
+  fetchData()
+})
+
+onUnmounted(() => {
+  purgeAllCharts()
+  dataStore.apiData[endpoint] = null
+  dataStore.dataErrors[endpoint] = false
+})
+
+const isLoading = computed(
+  () => !!latLng.value && !apiData.value && !dataError.value
+)
+</script>
+
+<template>
+  <div class="xray-wrapper">
+    <div v-if="latLng">
+      <Era5WrfXrayControls
+        v-model:startDate="startDate"
+        v-model:endDate="endDate"
+      />
+
+      <div v-if="dataError" class="notification is-danger mt-4">
+        Unable to load data for this location. Try a different location.
+      </div>
+
+      <div v-else-if="isLoading" class="mt-4">
+        <div class="loading-state">
+          <progress class="progress is-small is-primary" max="100">
+            Loading
+          </progress>
+          <span>Fetching downscaled ERA5 reanalysis time series…</span>
+        </div>
+      </div>
+
+      <div v-else-if="apiData" class="mt-4">
+        <!-- Temperature -->
+        <div class="mb-5">
+          <Era5WrfTemperatureChart
+            :t2Max="seriesByVariable.t2_max"
+            :t2Mean="seriesByVariable.t2_mean"
+            :t2Min="seriesByVariable.t2_min"
+            :lat="latLng.lat"
+            :lng="latLng.lng"
+          />
+        </div>
+
+        <!-- Humidity -->
+        <div class="mb-5">
+          <Era5WrfHumidityChart
+            :rh2Max="seriesByVariable.rh2_max"
+            :rh2Mean="seriesByVariable.rh2_mean"
+            :rh2Min="seriesByVariable.rh2_min"
+            :lat="latLng.lat"
+            :lng="latLng.lng"
+          />
+        </div>
+
+        <!-- Precipitation -->
+        <div class="mb-5">
+          <Era5WrfPrecipitationChart
+            :rainnc="seriesByVariable.rainnc_sum"
+            :lat="latLng.lat"
+            :lng="latLng.lng"
+          />
+        </div>
+
+        <!-- Wind Rose -->
+        <div>
+          <Era5WrfWindRose
+            :speedSeries="seriesByVariable.wspd10_mean"
+            :directionSeries="seriesByVariable.wdir10_mean"
+            speedLabel="Mean Wind Speed"
+            chartId="wind-rose-mean"
+          />
+        </div>
+      </div>
+    </div>
+
+    <div v-else-if="props.showLocationPrompt" class="notification is-info">
+      Select a location to query the downscaled ERA5 reanalysis dataset.
+    </div>
+  </div>
+</template>
+
+<style scoped>
```

Comment:
Remove styles related to loading state, likely meld into Gimme pending untangling loading stuff